internal/controller/nimcache_controller.go

```diff
@@ -1194,14 +1192,33 @@ func (r *NIMCacheReconciler) constructJob(ctx context.Context, nimCache *appsv1a
 
 		// Inject custom CA certificates when running in a proxy envronment
 		if nimCache.Spec.CertConfig != nil {
-			volumes, volumeMounts, err := r.createCertVolumesAndMounts(ctx, nimCache)
-			if err != nil {
-				logger.Error(err, "failed to create volume mounts for custom CA cert injection")
-				return nil, err
-			}
+			logger.Error(err, "Deprecated field 'CertConfig' is used. Please migrate to 'Proxy' field on NIMCache.\"")
+			return nil, err
+		}
 
-			job.Spec.Template.Spec.Volumes = append(job.Spec.Template.Spec.Volumes, volumes...)
-			job.Spec.Template.Spec.Containers[0].VolumeMounts = append(job.Spec.Template.Spec.Containers[0].VolumeMounts, volumeMounts...)
+		if nimCache.GetProxySpec() != nil {
+			job.Spec.Template.Spec.InitContainers = nimCache.GetInitContainers()
+			job.Spec.Template.Spec.Containers[0].Env = utils.MergeEnvVars(job.Spec.Template.Spec.Containers[0].Env, nimCache.GetEnvWithProxy())
+			job.Spec.Template.Spec.Containers[0].VolumeMounts = append(job.Spec.Template.Spec.Containers[0].VolumeMounts,
+				corev1.VolumeMount{
+					Name:      "ca-cert-volume",
+					MountPath: "/etc/ssl",
+				},
+			)
+			job.Spec.Template.Spec.Volumes = append(job.Spec.Template.Spec.Volumes,
```

Comment:
we can move the spec for these volumes into a common method between nimcache and nimservice and `ca-cert-volume` as a shared constant.

Reply:
Updated this!